pkg/migrations/sql/idempotent/base.sql

```diff
@@ -1983,6 +2031,71 @@ SET search_path = pg_temp;
 REVOKE ALL ON FUNCTION SCHEMA_CATALOG.create_metric_view(text) FROM PUBLIC;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.create_metric_view(text) TO prom_writer;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.register_metric_view(schema_name name, view_name name, if_not_exists BOOLEAN = false)
+    RETURNS BOOLEAN
+AS $func$
+DECLARE
+   metric_table_name name;
+   column_count int;
+BEGIN
+    -- check if table/view exists
+    PERFORM * FROM information_schema.tables 
+    WHERE  table_schema = register_metric_view.schema_name
+    AND    table_name   = register_metric_view.view_name;
+
+    IF NOT FOUND THEN
+        RAISE EXCEPTION 'cannot register non-existant metric view in specified schema';
+    END IF;
+
+    -- cannot register view in data schema
+    IF schema_name = 'SCHEMA_DATA' THEN
+        RAISE EXCEPTION 'cannot register metric view in data schema';
+    END IF; 
+
+    -- check if view is based on a metric from prom_data
+    SELECT table_name FROM information_schema.view_table_usage
+    INTO metric_table_name
+    WHERE view_schema = register_metric_view.schema_name
+    AND view_table_usage.view_name = register_metric_view.view_name
+    AND table_schema = 'SCHEMA_DATA';
+
+    IF NOT FOUND THEN
+        RAISE EXCEPTION 'view not based on a metric table from SCHEMA_DATA schema';
+    END IF;
+
+    -- check if the view contains necessary columns with the correct types
+    SELECT count(*) FROM information_schema.columns
+    INTO column_count
+    WHERE table_schema = register_metric_view.schema_name
+    AND table_name   = register_metric_view.view_name
+    AND ((column_name = 'time' AND data_type = 'timestamp with time zone')
+    OR (column_name = 'series_id' AND data_type = 'bigint')
+    OR (column_name = 'value' AND data_type = 'double precision'));
+
+    IF column_count <> 3 THEN
+        RAISE EXCEPTION 'view must contain time, series_id, and value columns with appropriate types';
+    END IF;
+
+    -- insert into metric table
+    INSERT INTO SCHEMA_CATALOG.metric (metric_name, table_name, table_schema, series_table, is_view, creation_completed)
+    VALUES (register_metric_view.view_name, register_metric_view.view_name, register_metric_view.schema_name, metric_table_name, true, true)
+    ON CONFLICT DO NOTHING;
+
+    IF NOT FOUND AND NOT register_metric_view.if_not_exists THEN
```

Comment:
I think if if_not_exists is true we should raise a NOTICE  to let people know it already existed. That's standard practice in PG